### ISSUE TITLE
Feature request: compatibility with sf

### ISSUE THREAD

**Issue by taoofcoffee** [ctmm-initiative/ctmm#37]
At present, ctmm allows the export of UDs as a raster, SpatialPolygonsDataFrame, or as a shapefile. Considering the movement in the spatial community to use sf instead of sp, it would be great if ctmm had compatibility with this package in the future.

**Comment by chfleming** [comment]
Just glancing at `sf`, would there be anything lost if the coercion functions went `ctmm -> sp -> sf` ?

**Comment by taoofcoffee** [comment]
That's a good question. I found [this information](https://r-spatial.github.io/sf/articles/sf1.html#conversion-including-to-and-from-sp-1) regarding conversion from sp to sf:

"As the Spatial* objects only support MULTILINESTRING and MULTIPOLYGON, LINESTRING and POLYGON geometries are automatically coerced into their MULTI form. When converting Spatial* into sf, if all geometries consist of a single POLYGON (possibly with holes), a POLYGON and otherwise all geometries are returned as MULTIPOLYGON: a mix of POLYGON and MULTIPOLYGON (such as common in shapefiles) is not created. Argument forceMulti=TRUE will override this, and create MULTIPOLYGONs in all cases. For LINES the situation is identical"

Given that sp objects exported from UD objects are either single or multipolygons, it seems like it should be an easy conversion. The only problem I've encountered converting sp to sf has been with multiline files that result in invalid geometries, but that should not be the case here.
